### ISSUE TITLE
False positive - eslint-plugin-react(jsx-key): Missing "key" prop for element in array.

### ISSUE THREAD

**Issue by maurice** [oxc-project/oxc#1857]
Defining a [decorator for a Ladle story](https://ladle.dev/docs/decorators/) like so...

```
MyStory.decorators = [
  (Component) => (
    <div style={{ margin: "3em" }}>
      <Component />
    </div>
  ),
];
```

reports that the `<div>` should have a key. I believe this is incorrect because this is actually an anonymous React FC that happens itself to be within an array, but the `<div>` is not like this

```
⚠ eslint-plugin-react(jsx-key): Missing "key" prop for element in array.
   ╭─[jsx_key.tsx:2:1]
 2 │         MyStory.decorators = [
 3 │           (Component) => <div><Component /></div>
   ·                           ───
 4 │         ];
   ╰────
```

I would love to contribute a fix for this myself... please let me try to provide a PR...
